examples/cluster/plot_inductive_clustering.py

```diff
@@ -0,0 +1,98 @@
+datapoints. Recomputing the clusters everytime we receive some new data
+is thus in many cases, intractable. With more data, there is also the
+possibility of degrading the previous clustering.
+One solution to this problem, is to first infer the target classes using
+some unsupervised learning algorithm and then fit a classifier on the
+inferred targets, treating it as a supervised problem.
+"""
+print(__doc__)
+
+import matplotlib.pyplot as plt
+from sklearn.base import BaseEstimator, clone
+from sklearn.cluster import AgglomerativeClustering
+from sklearn.datasets import make_blobs
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.utils.metaestimators import if_delegate_has_method
+
+
+N_SAMPLES = 5000
+RANDOM_STATE = 42
+
+
+class InductiveClusterer(BaseEstimator):
+    def __init__(self, clusterer, classifier):
+        self.clusterer = clusterer
+        self.classifier = classifier
+
+    def fit(self, X, y=None):
+        self.clusterer_ = clone(self.clusterer)
+        self.classifier_ = clone(self.classifier)
+        y = self.clusterer_.fit_predict(X)
+        self.classifier_.fit(X, y)
+        return self
+
+    @if_delegate_has_method(delegate='classifier')
+    def predict(self, X):
+        return self.classifier_.predict(X)
+
+    @if_delegate_has_method(delegate='classifier')
+    def decision_function(self, X):
+        return self.classifier_.decision_function(X)
+
+
+def plot_scatter(X,  color, alpha=0.5):
+    return plt.scatter(X[:, 0],
+                       X[:, 1],
+                       c=color,
+                       alpha=alpha,
+                       edgecolor='k')
+
+
+# Generate some training data from clustering
+X, y = make_blobs(n_samples=N_SAMPLES,
+                  cluster_std=[1.0, 1.0, 0.5],
+                  centers=[(-5, -5), (0, 0), (5, 5)],
+                  random_state=RANDOM_STATE)
+
+
+# Train a clustering algorithm on the training data and get the cluster labels
+clusterer = AgglomerativeClustering(n_clusters=3)
+cluster_labels = clusterer.fit_predict(X)
+
+plt.subplot(131)
+plot_scatter(X, cluster_labels)
+plt.title("Ward Linkage")
+
+
+# Generate new samples and plot them along with the original dataset
+X_new, y_new = make_blobs(n_samples=10,
+                          centers=[(-7, -1), (-2, 4), (3, 6)],
+                          random_state=RANDOM_STATE)
+
+plt.subplot(132)
+plot_scatter(X, cluster_labels)
+plot_scatter(X_new, 'black', 1)
+plt.title("Ward Linkage and unknown instances")
+
+
+# Declare the inductive learning model that it will be used to
+# predict cluster membership for unknown instances
+classifier = RandomForestClassifier(random_state=RANDOM_STATE)
+inductiveLearner = InductiveClusterer(clusterer, classifier).fit(X)
+
+probable_clusters = inductiveLearner.predict(X_new)
+
+
+plt.subplot(133)
+plot_scatter(X, cluster_labels)
+plot_scatter(X_new, probable_clusters)
+plt.title("Inductive inference on cluster membership \n"
```

Comment:
The title is too long for the plot.
Please make sure the final plot is readable with the chosen figure size.

Reply:
I shorted the titles